### PR TITLE
Support asynchronous and responseless pipelining

### PR DESCRIPTION
**Pull request by g-andrade**

### CODE REVIEW

**Review by lrascao — APPROVED** [review]
👍 could you please squash redundant commits?

**Comment by g-andrade** [comment]
> could you please squash redundant commits?

Done.

**Review by paulo-ferraz-oliveira — APPROVED** [review]
I approve this

File: src/eredis_client.erl [hunk 1]
```diff
@@ -267,7 +284,6 @@ reply(Value, Queue) ->
 
 %% @doc Send `Value' to each client in queue. Only useful for sending
 %% an error message. Any in-progress reply data is ignored.
--spec reply_all(any(), eredis_queue()) -> ok.
```
Comment (paulo-ferraz-oliveira) [inline src/eredis_client.erl:270]: Why?

**Pull request merged**